app/src/main/java/mollie/tictactoe/MainActivity.java

```diff
@@ -14,14 +14,18 @@
 
 public class MainActivity extends AppCompatActivity {
 
-    private List<Button> mBoard = new ArrayList<>();
+    private final List<Button> mBoard = new ArrayList<>();
     private GameEngine mGameEngine;
+    private static final String EXTRA_GAME_TYPE = "tictactoe.game_type";
+    private static final String EXTRA_BOARD_TYPE = "tictactoe.board_type";
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-        createGame();
+        String[] players = getIntent().getStringArrayExtra(EXTRA_GAME_TYPE);
```

Comment:
@felipesere I agree my naming isn't good here, and will change it.  I'm a little of unsure of how to abstract player types out of the extra though.  I am going to use my PlayerFactory to create the Players when I have more options, but I feel that I will still have to pass strings between my activities in order to create the correct player types?  I'm not sure how I would pass objects between the activities, if that is what you are suggesting?  Or just to have an intermediary class that can receive the strings and return player instances?